### PR TITLE
Add Junit XML report

### PR DESCRIPTION
**Pull request by marob**
Junit XML is a standard report format. It can then be used in CI platforms (Jenkins for example).

In order not to fail the build (and to be able to publish a report in the CI platform), I've also added the possibility to deactivate the threshold (with a negative value).

### CODE REVIEW

**Comment by dylanb** [comment]
@marob this is an awesome contribution. Could you add some tests?

**Comment by marob** [comment]
@dylanb done!

File: test/unit/axe-webdriver.js [hunk 1]
```diff
@@ -18,14 +18,4 @@ describe('axe-webdriver grunt task', function () {
 		call.args[0].should.equal('axe-webdriver');
 		(typeof call.args[2]).should.equal('function');
 	});
-	it('should register an axe-webdriver task with callback function', function () {
```
Comment (dylanb) [inline test/unit/axe-webdriver.js]: Why did you delete this test?

Comment (marob) [inline test/unit/axe-webdriver.js]: Because it's a duplicate of the previous one.

File: README.md [hunk 2]
```diff
@@ -51,6 +52,7 @@ Type: `Number`
 Default value: `0`
 
 A number that represents the maximum number of allowable violations. Each violation represents a rule that fails, it may fail for an number of nodes. It is recommended that this value not be changed.
+A negative value will prevent failure whatever the number of violations.
```
Comment (dylanb) [inline README.md]: Need a test for this change

File: test/unit/junitReporter.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
+var junitReportBuilder = require('junit-report-builder');
```
Comment (dylanb) [inline test/unit/junitReporter.js]: I think it would be good to have tests to ensure that the file is being created properly. What happens if the file already exists? We should test that the correct behavior happens for an existing file.

File: README.md (hunk 2 above)
Comment (marob) [inline README.md]: I'll add a test for this.

File: test/unit/junitReporter.js (hunk 3 above)
Comment (marob) [inline test/unit/junitReporter.js]: Are you talking about the produced XML file? As I'm using the junit-report-builder module for the XML creation & file system write, it is not our task to test it: the junit-report-builder module should have such tests.

File: README.md (hunk 2 above)
Comment (marob) [inline README.md]: I've added the test.

**Comment by marob** [comment]
I've added the test about the negative value to prevent failure.
Is it OK now?

File: test/unit/repoter.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
+var reporter = require('../../lib/reporter.js');
```
Comment (dylanb) [inline test/unit/repoter.js]: this is a really small, nitpicking issue, but you mis-spelled the name of the file - please fix :-)

Comment (marob) [inline test/unit/repoter.js]: Yep, you're right. It's fixed.
